docs/src/main/asciidoc/building-native-image.adoc

```diff
@@ -14,7 +14,50 @@ This guide covers:
 
 This guide takes as input the application developed in the link:getting-started[Getting Started Guide].
 
-== Prerequisites
+== GraalVM
+
+Building a native executable requires using a distribution of GraalVM.
+There are two distributions: Oracle GraalVM and Mandrel.
+The differences between the two distributions are as follows:
+
+* Mandrel is a downstream distribution of the Oracle GraalVM community edition.
+Mandrel's main goal is to provide a way to build native executables specifically designed to support Quarkus.
+
+* Mandrel releases are built from a code base derived from the upstream Oracle GraalVM code base,
+with only minor changes but some significant exclusions.
+They support the same capabilities to build native executables as Oracle GraalVM,
+with no significant changes to functionality.
+Notably, they do not include support for polyglot programming.
+
+* Mandrel is built slightly differently to GraalVM, using the standard OpenJDK project.
+This means that it does not profit from a few small enhancements that Oracle have added to the version of OpenJDK used to build their own GraalVM downloads.
+
+* Mandrel is only available for projects that can be compiled with Java 11.
+If you are using Java 8,
+you should consider using Oracle GraalVM instead.
```

Comment:
I read all these points and I only see drawbacks for Mandrel: it has less features, doesn't have some enhancements and doesn't support Java 8. It's a bit of a hard sell :).

Reply:
/me chipping in from my vacation: The fact that something is *smaller* and *focused* on a particular function is IMHO a huge plus. I am not a copywriter, yet I think we should formulate something to the fact that Mandrel is minimized and focused. In the age of bloatware where a simple web page pushes 10MB of data you didn't want on you, there is a value in delivering less.

Reply:
+1 we should point out that mandrel distribution is ~200MB while graalvm CE is ~400MB.

Reply:
@gsmet This is not at all about us providing an inferior product. It is about us providing a better level of support to the majority of our users.

mandrel removes a bundle of GraalVM features that are not needed for the vast majority of Quarkus native apps. It omits OpenJDK 'enhancements' to JDK11 that our OpenJDK team does not manage and cannot vouch for (that's especially important when it comes to conformance and security). It does not attempt to work on ljdk8 but that's a legacy release of OpenJDK which most Quarkus users will have no good reason to adopt and for which, in consequence, we have no commitment to support Quarkus beyond best effort. In other words, mandrel streamlines and simplifies the build, test, release and support cycle for most of our users, especially those who pay for support.

Note that we are not forcing anyone's hand here. We won't refuse to support anyone using Quarkus on GraalVM for preference whether with a labs jdk11u or labs jdk8u. Indeed, we will do or best to provide support for any problems that manifest in GraalVM when using that configuration. Obviously, we would much prefer if our paid support customers were to use mandrel. If they don't then should any problems manifest on GraalVM we will almost certainly want them to check -- where possible -- that those  problems also reproduce on mandrel. If they cannot then we will still try to help but we are not going to write a blank cheque based on whatever code Oracle labs might add to GraalVM beyond the native image component. Anyway, the idea that we are pushing some inferior variant at the expense of GraalVM is very far from the truth.